src/metabase/lib/metadata/jvm.clj

```diff
@@ -25,12 +25,25 @@
   (or (qualified-keyword? k)
       (str/includes? k ".")))
 
+(def ^:private memoized-kebab-key
+  "Calculating the kebab-case version of a key every time is pretty slow (even with the LRU
+  caching [[u/->kebab-case-en]] has), since the keys here are static and finite we can just memoize them forever with
+  regular memoization and get a nice performance boost."
+  ;; actually, just using a simple atom like this is 10x faster than [[memoize]], and thus 100x faster
+  ;; than [[metabase.util.memoize/lru]]... see https://metaboat.slack.com/archives/C04CYTEL9N2/p1702610853943109
+  (let [cache (atom (hash-map))]
+    (fn [k]
+      (or (get @cache k)
+          (let [k' (u/->kebab-case-en k)]
+            (swap! cache assoc k k')
+            k')))))
```

Comment:
I spent a lot of time wringing my hands trying to decide if `instance->metadata` below was going to be too slow to use in hydration code vs just manually building a map with the three keys I needed, eventually I decided that speeding up `u/->kebab-case-en` in this way gave us a significant performance boost and could let me use `instance->metadata` without worrying about the performance too much. See https://metaboat.slack.com/archives/C04CYTEL9N2/p1702610853943109 for more info.

Reply:
Have you considered using a `ConcurrentHashMap` with `computeIfAbsent` instead of the atom?